openeogeotrellis/layercatalog.py

```diff
@@ -800,6 +800,93 @@ def _native_crs(self, metadata: GeopysparkCubeMetadata) -> str:
+        outer_bounds = [None, None]
+        if catalog_temporal_extent:
+            for extent in catalog_temporal_extent:
+                if extent[0]:
+                    if outer_bounds[0] is None:
+                        outer_bounds[0] = extent[0]
+                    else:
+                        outer_bounds[0] = min(outer_bounds[0], extent[0])
+                if extent[1]:
+                    if outer_bounds[1] is None:
+                        outer_bounds[1] = extent[1]
+                    else:
+                        outer_bounds[1] = max(outer_bounds[1], extent[1])
+        if temporal_extent_constraints is None:
+            temporal_extent = outer_bounds
+        else:
+            # take the intersection of outer_bounds and temporal_extent
+            beginnings = []
+            if outer_bounds[0]:
+                beginnings.append(outer_bounds[0])
+            if temporal_extent_constraints[0]:
+                beginnings.append(temporal_extent_constraints[0])
+            if not beginnings:
+                beginnings.append(None)
+
+            ends = []
+            if outer_bounds[1]:
+                ends.append(outer_bounds[1])
+            if temporal_extent_constraints[1]:
+                ends.append(temporal_extent_constraints[1])
+            if not ends:
+                ends.append(None)
+
+            temporal_extent = (
+                max(beginnings),  # ISO date is sortable like a string
+                min(ends),
+            )
+        return temporal_extent
+
+    def estimate_number_of_temporal_observations(self,
+                                                 collection_id: str,
+                                                 constraints: dict,
+                                                 ) -> int:
+        temporal_extent = self.derive_temporal_extent(collection_id, constraints)
+
+        metadata_json = self.get_collection_metadata(collection_id=collection_id)
+        metadata = GeopysparkCubeMetadata(metadata_json)
+
+        consider_as_singular_time_step = deep_get(metadata_json, "_vito", "data_source",
+                                                  "consider_as_singular_time_step", default=False)
+        if consider_as_singular_time_step:
+            return 1
+
+        # step could be explicitly 'None', so we use 'or' to specify the default
+        temporal_step = metadata.get("cube:dimensions", "t", "step", default=None) or "P10D"
+
+        # https://github.com/stac-extensions/datacube?tab=readme-ov-file#temporal-dimension-object
+        temporal_step = parse_approximate_isoduration(temporal_step)
+        estimate_days_per_sample = temporal_step.days
+
+        from_date, to_date = temporal_extent
+        if to_date is None:
+            if from_date is None:
+                days = 1
+                logger.warning(
+                    f"Got open temporal extent: {repr(temporal_extent)}. Assuming {days} day."
```

Comment:
Python pro-tip: use the [`=` specifier in f-strings](https://docs.python.org/3/whatsnew/3.8.html#f-strings-support-for-self-documenting-expressions-and-debugging) to simplify these kind of logging constructs.

for example:
```suggestion
                    f"Got open {temporal_extent=}. Assuming {days=}."
```